src/lib/hooks/use-query-params.ts

```diff
@@ -0,0 +1,71 @@
+'use client'
+
+import { Token } from '@indexcoop/flash-mint-sdk'
+import { useSearchParams } from 'next/navigation'
+import { useMemo } from 'react'
+
+import { getCurrencyTokens, getLeverageTokens } from '@/app/leverage/constants'
+import { LeverageToken, LeverageType } from '@/app/leverage/types'
+import { chains } from '@/lib/utils/wagmi'
+
+type DefaultParams = {
+  isMinting: boolean
+  leverageType: LeverageType
+  network: number
+  inputToken: Token
+  outputToken: Token
+}
+
+type ReturnType<T extends Partial<DefaultParams>> = {
+  [K in keyof DefaultParams as `query${Capitalize<string & K>}`]: K extends keyof T
+    ? T[K]
+    : DefaultParams[K] | undefined
+}
+
+export const useQueryParams = <T extends Partial<DefaultParams>>(
+  defaultParams: T = {} as T,
+): ReturnType<T> => {
+  const searchParams = useSearchParams()
+
+  return useMemo(() => {
+    const network = parseInt(searchParams.get('network') || '0')
+    const buy = searchParams.get('buy') || ''
+    const sell = searchParams.get('sell') || ''
+
+    const queryNetwork =
+      chains.find((chain) => chain.id === network)?.id ?? defaultParams.network
+
+    const tokens = getCurrencyTokens(queryNetwork ?? 0).concat(
+      getLeverageTokens(network),
+    )
+
+    const queryOutputToken =
+      tokens.find(
+        (token) => token.symbol.toLowerCase() === buy.toLowerCase(),
+      ) ?? defaultParams.outputToken
+    const queryInputToken =
+      tokens.find(
+        (token) => token.symbol.toLowerCase() === sell.toLowerCase(),
+      ) ?? defaultParams.inputToken
```

Comment:
This will need to have a more sophisticated logic e.g. testing if one token is an index and the other a currency. 

Otherwise, you'll get weird results if someone would do things like: 
* currency tokens for buy and sell: `/leverage?network=1&sell=eth&buy=eth`
* non existant index tokens for that chain: `leverage?network=1&sell=eth3x&buy=eth`

Reply:
it does handle that

Reply:
It should be good now